src/shogun/base/SGObject.h

```diff
@@ -1022,5 +997,187 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+	template <class T>
+	class ObservedValueTemplated;
+
+	/**
+	 * Observed value which is emitted by algorithms.
+	 */
+	class ObservedValue : public CSGObject
```

Comment:
@karlnapf as I mentioned before, I think the best way to solve the cyclic dependency problem is to add the ObservedValue class here in SGObject.h. By just adding 100~ lines more we can easily use the observable utilities (and we may be able to get rid of all those `make_observation` methods). @vigsterkr any opinion on this?